pkg/machinery/config/types/v1alpha1/v1alpha1_etcdconfig.go

```diff
@@ -0,0 +1,44 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package v1alpha1
+
+import (
+	"fmt"
+	goruntime "runtime"
+
+	"github.com/talos-systems/crypto/x509"
+
+	"github.com/talos-systems/talos/pkg/machinery/constants"
+)
+
+// Image implements the config.Etcd interface.
+func (e *EtcdConfig) Image() string {
+	image := e.ContainerImage
+	suffix := ""
+
+	if goruntime.GOARCH == "arm64" {
+		suffix = "-arm64"
+	}
+
+	if image == "" {
+		image = fmt.Sprintf("%s:%s%s", constants.EtcdImage, constants.DefaultEtcdVersion, suffix)
+	}
+
+	return image
+}
+
+// CA implements the config.Etcd interface.
+func (e *EtcdConfig) CA() *x509.PEMEncodedCertificateAndKey {
+	return e.RootCA
+}
+
+// ExtraArgs implements the config.Etcd interface.
+func (e *EtcdConfig) ExtraArgs() map[string]string {
+	if e.EtcdExtraArgs == nil {
+		e.EtcdExtraArgs = make(map[string]string)
```

Comment:
probably not for this PR, but we should review places like this one (writing to config object) and remove them, as it's a perfect data race